src/telemetry/telemetry.bif

```diff
@@ -548,3 +558,36 @@ function Telemetry::__dbl_histogram_sum%(val: opaque of dbl_histogram_metric%):
 		return zeek::make_intrusive<DoubleVal>(0.0);
 		}
 	%}
+
+
+function Telemetry::__collect_metrics%(prefix: string, name: string%): any_vec
+	%{
+	auto metrics = telemetry_mgr->CollectMetrics(sv(prefix), sv(name));
+
+	static auto metrics_vector_type = zeek::id::find_type<VectorType>("any_vec");
+	auto vec = zeek::make_intrusive<VectorVal>(metrics_vector_type);
+
+	for ( const auto& m : metrics )
+		{
+		auto rec = m.GetMetricRecord();
+		vec->Append(rec);
+		}
+
+	return vec;
+	%}
+
+function Telemetry::__collect_histogram_metrics%(prefix: string, name: string%): any_vec
+	%{
+	auto metrics = telemetry_mgr->CollectHistogramMetrics(sv(prefix), sv(name));
+
+	static auto metrics_vector_type = zeek::id::find_type<VectorType>("any_vec");
```

Comment:
Same question here. By having `static` variables in two different functions, we ask the compiler for two complete sets of synchronization primitives to cache the same thing.

Reply:
See above.